lib/internal/process/warning.js

```diff
@@ -13,26 +78,44 @@ function setupProcessWarnings() {
       const trace = process.traceProcessWarnings ||
                     (isDeprecation && process.traceDeprecation);
       if (trace && warning.stack) {
-        console.error(`${prefix}${warning.stack}`);
+        if (warning.code) {
+          output(`${prefix}[${warning.code}] ${warning.stack}`);
+        } else {
+          output(`${prefix}${warning.stack}`);
+        }
       } else {
-        var toString = warning.toString;
-        if (typeof toString !== 'function')
-          toString = Error.prototype.toString;
-        console.error(`${prefix}${toString.apply(warning)}`);
+        const toString =
+          typeof warning.toString === 'function' ?
+            warning.toString : Error.prototype.toString;
+        if (warning.code) {
+          output(`${prefix}[${warning.code}] ${toString.apply(warning)}`);
+        } else {
+          output(`${prefix}${toString.apply(warning)}`);
+        }
       }
     });
   }
 
   // process.emitWarning(error)
-  // process.emitWarning(str[, name][, ctor])
-  process.emitWarning = function(warning, name, ctor) {
-    if (typeof name === 'function') {
-      ctor = name;
-      name = 'Warning';
+  // process.emitWarning(str[, type[, code]][, ctor])
+  process.emitWarning = function(warning, type, code, ctor) {
+    if (typeof type === 'function') {
+      ctor = type;
+      code = undefined;
+      type = 'Warning';
+    }
+    if (typeof code === 'function') {
+      ctor = code;
+      code = undefined;
     }
+    if (code !== undefined && typeof code !== 'string')
+      throw new TypeError('`code` must be a String');
```

Comment:
I thought we were using single quotes around parameter names instead of backticks in error messages?